src/components/EditFeatureDrawer/EditFeatureFullForm/index.tsx

```diff
@@ -0,0 +1,223 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState
+} from 'react';
+
+import {
+  Alert
+} from 'antd';
+import {
+  useForm
+} from 'antd/lib/form/Form';
+
+import {
+  Feature
+} from 'geojson';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import moment from 'moment';
+
+import {
+  useTranslation
+} from 'react-i18next';
+
+import {
+  Logger
+} from '@terrestris/base-util';
+
+import MapUtil from '@terrestris/ol-util/dist/MapUtil/MapUtil';
+
+import useMap from '@terrestris/react-geo/dist/Hook/useMap';
+
+import {
+  WmsLayer,
+  isWmsLayer
+} from '@terrestris/react-geo/dist/Util/typeUtils';
+
+import {
+  PropertyFormItemEditConfig,
+  PropertyFormTabConfig
+} from '@terrestris/shogun-util/dist/model/Layer';
+
+import useAppDispatch from '../../../hooks/useAppDispatch';
+import useAppSelector from '../../../hooks/useAppSelector';
+import useExecuteGetFeature from '../../../hooks/useExecuteGetFeature';
+import useSHOGunAPIClient from '../../../hooks/useSHOGunAPIClient';
+
+import {
+  setFeature
+} from '../../../store/editFeature';
+
+import EditFeatureGeometryToolbar from '../EditFeatureGeometryToolbar';
+import EditFeatureTabs from '../EditFeatureTabs';
+import EditFeatureToolbar from '../EditFeatureToolbar';
+
+export type EditFeatureFullFormProps = {
+  feature: Feature;
+  layer: WmsLayer;
+};
+
+export const EditFeatureFullForm: React.FC<EditFeatureFullFormProps> = ({
+  feature,
+  layer
+}) => {
+  const {
+    t
+  } = useTranslation();
+  const executeGetFeature = useExecuteGetFeature();
+
+  const [tabConfig, setTabConfig] = useState<PropertyFormTabConfig<PropertyFormItemEditConfig>[]>();
+  const [initialValues, setInitialValues] = useState<Record<string, any>>();
+  const [errorMsg, setErrorMsg] = useState<string>();
+
+  const allowedEditMode = useAppSelector(
+    state => state.editFeature.userEditMode
+  );
+
+  const [form] = useForm();
+  const map = useMap();
+  const dispatch = useAppDispatch();
+  const client = useSHOGunAPIClient();
+
+  const reloadFeature = useCallback(async (id: string) => {
+    if (!layer || !isWmsLayer(layer)) {
+      return;
+    }
+
+    const updatedFeatures = await executeGetFeature({
+      layer: layer,
+      featureId: id
+    });
+
+    if (
+      updatedFeatures?.features[0] &&
+      (allowedEditMode.includes('UPDATE') ||
+      allowedEditMode.includes('DELETE'))
```

Comment:
Why do we check for the roles here? I would assume it needs to be done _before_ the user actually can trigger updating/deleting a feature?

Reply:
The check is obsolete at this point and has been removed.